fe/fe-core/src/main/java/com/starrocks/external/iceberg/IcebergHiveCatalog.java

```diff
@@ -133,9 +139,48 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
         throw new UnsupportedOperationException("Not implemented");
     }
 
+    public List<String> listAllDatabases() {
+        try {
+            return new ArrayList<>(clients.run(IMetaStoreClient::getAllDatabases));
+        } catch (TException e) {
+            throw new RuntimeException(e);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public Database getDB(String dbName) {
+        try {
+            org.apache.hadoop.hive.metastore.api.Database db = clients.run(client -> client.getDatabase(dbName));
+            if (db == null || db.getName() == null) {
+                throw new TException("Hive db " + dbName + " doesn't exist");
+            }
+        } catch (TException e) {
+            throw new RuntimeException(e);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        return convertToSRDatabase(dbName);
+    }
+
     @Override
     public List<TableIdentifier> listTables(Namespace namespace) {
-        throw new UnsupportedOperationException("Not implemented");
+        String database = namespace.level(0);
+        try {
+            List<String> tableNames = clients.run(client -> client.getAllTables(database));
```

Comment:
I think just return tableNames. What is the purpose of the following operations.

Reply:
It's to verify whether the tables are Iceberg table. Just return tableNames will get all tables in metastore.